content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,389 @@
+    IKS 37 days before they expire.
+    You will have to recreate the secrets by rerunning the instructions of this section every time the secrets provided
+    by IKS are updated. You may want to use scripts or operators to automate this and keep the
+    secrets in sync.
+    {{< /warning >}}
+
+    {{< text bash >}}
+    $ kubectl create -n istio-system secret tls istio-ingressgateway-certs --key ingress_gateway_certs/tls.key --cert trusted.crt
+    $ kubectl create -n istio-system secret generic istio-ingressgateway-ca-certs --from-file=trustid-x3-root.pem
+    $ kubectl create secret generic alb-certs -n istio-system --from-file=trusted.crt --from-file=alb_certs/client.crt --from-file=alb_certs/client.key
+    secret "istio-ingressgateway-certs" created
+    secret "istio-ingressgateway-ca-certs" created
+    secret "alb-certs" created
+    {{< /text >}}
+
+## Configure a mutual TLS ingress gateway
+
+In this section you configure the Istio ingress gateway to perform mutual TLS between external clients and the gateway.
+You use the certificates and the keys provided to you for the ingress gateway and ALB.
+
+1.  Define a `Gateway` to allow access on port 443 only, with mutual TLS:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: default-ingress-gateway
+    spec:
+      selector:
+        istio: ingressgateway # use istio default ingress gateway
+      servers:
+      - port:
+          number: 443
+          name: https
+          protocol: HTTPS
+        tls:
+          mode: MUTUAL
+          serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+          privateKey: /etc/istio/ingressgateway-certs/tls.key
+          caCertificates: /etc/istio/ingressgateway-ca-certs/trustid-x3-root.pem
+        hosts:
+        - "*"
+    EOF
+    {{< /text >}}
+
+1. Configure routes for traffic entering via the `Gateway`:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: default-ingress
+    spec:
+      hosts:
+      - "*"
+      gateways:
+      - default-ingress-gateway
+      http:
+      - match:
+        - uri:
+            prefix: /status
+        route:
+        - destination:
+            port:
+              number: 8000
+            host: httpbin.httptools.svc.cluster.local
+    EOF
+    {{< /text >}}
+
+1.  Delete the Istio Ingress Gateway's pod to reload the certificates:
+
+    {{< text bash >}}
+    $ kubectl delete pod -l istio=ingressgateway -n istio-system
+    {{< /text >}}
+
+1.  Send a request to `httpbin` by _curl_, passing as parameters the client certificate
+    (the `--cert` option) and the private key (the `--key` option):
+
+    {{< text bash >}}
+    $ curl https://$INGRESS_GATEWAY_DOMAIN/status/418 --cert alb_certs/client.crt  --key alb_certs/client.key
```

Comment:
is user supposed to run the curl cmd from their laptop or ssh into a pod?

Reply:
from the laptop, to access the service through the ingress gateway

Reply:
SG